internal/external/api/currency/abstract/abstract.go

```diff
@@ -0,0 +1,71 @@
+package abstract
+
+import (
+	"encoding/json"
+	"fmt"
+	"genesis-currency-api/pkg/dto"
+	"genesis-currency-api/pkg/errors"
+	"io"
+	"log"
+	"net/http"
+)
+
+//todo update docs
+
+type CurrencyRater interface {
+	GetCurrencyRate() (*dto.CurrencyResponseDTO, error)
+}
+
+type Client struct {
+	ApiURL       string
+	ProviderName string
+	Next         CurrencyRater
+}
+
+// ProcessCurrencyResponseDTO returns information about currency rate.
+func (s *Client) ProcessCurrencyResponseDTO(responseDTO *dto.CurrencyResponseDTO, err error) (*dto.CurrencyResponseDTO, error) {
+	if err == nil {
```

Comment:
Also, why it`s here?

Reply:
Sorry, didn't get your question?

Reply:
I mean, why are we error checking here, maybe it should be a previous level?

Reply:
Ah, I see. Because all raters should have a common logic of fetching data from the next API in case if an error happend with the currenct rater. So, I moved this "common" thing into an abstract rater.